### PR TITLE
[NP-7104] Add `foam.Color` library

### PR DESCRIPTION
**Pull request by KernelDeimos**

### CODE REVIEW

File: src/foam/color/colorlib.js [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+      cornsilk: "#fff8dc",
+      lemonchiffon: "#fffacd",
+      floralwhite: "#fffaf0",
+      snow: "#fffafa",
+      yellow: "#ffff00",
+      lightyellow: "#ffffe0",
+      ivory: "#fffff0",
+      white: "#ffffff"
+    },
+    COLOR_TO_NAME: { '#000000': 'black',
```
Comment (kgrgreer) [inline src/foam/color/colorlib.js]: Couldn't one of these just be a factory which generates based on the other to avoid sending data twice?

Comment (KernelDeimos) [reply src/foam/color/colorlib.js]: @kgrgreer how's this? I did it the metaprogramming way since foam.LIB constants don't have a factory property

**Review by sarthak-marwaha — APPROVED** [review]

File: src/foam/core/lib.js [hunk 2]
```diff
@@ -61,12 +61,20 @@ foam.LIB = function LIB(model) {
...
 
-    for ( var key in model.constants ) {
-      var v = root[key] = model.constants[key];
-      if ( foam.Object.isInstance(v) && v.class ) {
-        v = foam.lookup(v.class).create(v);
+    if ( Array.isArray(model.constants) ) {
+      for ( const constSpec of model.constants ) {
+        const constObj = foam.json.parse(constSpec, foam.core.Constant, foam.__context__);
+        if ( constObj.factory ) constObj.factory = constObj.factory.bind(root);
+        constObj.installInProto(root);
```
Comment (kgrgreer) [inline src/foam/core/lib.js]: But Constants don't exist when Lib is first defined, so why not just do: root[key] = v.value || v.factory.call(this);

Comment (KernelDeimos) [reply src/foam/core/lib.js]: simpler and works perfectly - thanks!

**Pull request merged**